test/legacyHomepageImageLoadingTest.js

```diff
@@ -0,0 +1,100 @@
+const puppeteer = require("puppeteer");
+
+function parseArgs(args) {
+  const parsedArgs = {};
+  for (let i = 0; i < args.length; i++) {
+    const arg = args[i];
+    if (arg.startsWith("--")) {
+      const keyValue = arg.split("=");
+      const key = keyValue[0].slice(2);
+      const value = keyValue.length > 1 ? keyValue[1] : true;
+      parsedArgs[key] = value;
+    }
+  }
+  return parsedArgs;
+}
+
+(async () => {
+  const args = process.argv.slice(2);
+  const parsedArgs = parseArgs(args);
+  //console.log('Parsed command line arguments:', parsedArgs);
+
+  const browser = await puppeteer.launch({ headless: true });
+  const page = await browser.newPage();
+
+  await page.setRequestInterception(true);
+
+  const loadedImageUrls = [];
+
+  // Intercept network requests
+  page.on("request", (request) => {
+    if (request.resourceType() === "image") {
+      // Log the URL of the image being requested
+      console.log("Image request:", request.url());
+      loadedImageUrls.push(request.url());
+    }
+    request.continue();
+  });
+
+  let homepage = "https://qa-digitalcollections.nypl.org";
```

Comment:
Everything before line 39 is the same in `test/homepageImageLoadingTest.js` and `test/homepageFeaturedImageLoadingTest.js. It'd be nice to see this refactored into one test script and the appropriate function (for legacy, swimlane, featured item) called so it's not three different files with similar code. Then, perhaps, the internal npm script can be 

```
    "homepageImageLoadingTestQa": "node test/imageLoadingTest.js --swimlane -- qa",
    "homepageImageLoadingTestProd": "node test/imageLoadingTest.js --swimlane --prod",
    "homepageFeaturedImageLoadingTestQa": "node test/imageLoadingTest.js --featuredImage -- qa",
    "homepageFeaturedImageLoadingTestProd": "node test/imageLoadingTest.js --featuredImage --prod"
    "legacyHomepageImageLoadingTestProd": "node test/imageLoadingTest.js --legacy --prod"
```

Reply:
I love this. If we want these tests to be maintained for the long term, we'll definitely do it.